include/cuco/static_set_ref.cuh

```diff
@@ -64,8 +64,16 @@ class static_set_ref
   : public detail::operator_impl<
       Operators,
       static_set_ref<Key, Scope, KeyEqual, ProbingScheme, StorageRef, Operators...>>... {
-  using impl_type =
-    detail::open_addressing_ref_impl<Key, Scope, KeyEqual, ProbingScheme, StorageRef>;
+  /// Flag indicating whether duplicate entries are allowed or not
+  static constexpr auto supports_duplicate_entries = false;
+
+  /// Implementation type
+  using impl_type = detail::open_addressing_ref_impl<Key,
+                                                     Scope,
+                                                     KeyEqual,
+                                                     ProbingScheme,
+                                                     StorageRef,
+                                                     supports_duplicate_entries>;
```

Comment:
Instead of exposing this to the public API, we could use a type selector in detail namespace that selects the appropriate input type based on the public type, e.g., `detail::impl_type<static_set_ref>`.
`detail::impl_type` would then specialize for `static_set_ref` and emit the OA ref impl class.

Reply:
This `impl_type` is private. Are you suggesting using a detail type selector to avoid exposing the private `impl_type` alias in a public header?

Reply:
Yeah that's basically it. It's just a minor remark and needs not to be dealt with in this PR.